### DISCUSSION TITLE
Make Dashboard Url unguessable

### DISCUSSION THREAD

**Discussion by lzm0** [apache/superset#19627] [Ideas]
**Is your feature request related to a problem? Please describe.**
When I create a dashboard its URL defaults to something like http://localhost:8088/superset/dashboard/1/, even if `PERMALINK_KEY_TYPE` was set to UUID. If I use the permalink, I would still be redirected to the short version of the URL.

**Describe the solution you'd like**
Use the UUID-based URL by default.

**Describe alternatives you've considered**
Turn off redirect of permalinks.

**Additional context**
For example, I have configured the dashboard to be public and
https://localhost:8088/superset/dashboard/p/X3NG2qWODon/
redirects to 
https://localhost:8088/superset/dashboard/1/?permalink_key=X3NG2qWODon
This makes dashboard URLs potentially guessable.

**Comment by srinify** [comment]
Thanks for the suggestion! Moving to Discussions so we can keep Issues focused on bugs and sips!

**Comment by lzm0** [comment]
I also don't understand why would I still get permalinks like `superset/dashboard/p/X3NG2qWODon` even though `PERMALINK_KEY_TYPE` is `uuid` by default.

**Comment by GerbenvdHuizen** [comment]
We currently have the same issue that users can potentially guess other public dashboard URLs even though initially a permalink was provided. Would be nice if there was no redirect or the permalink ID was used instead.
